File: 283. Move Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        num = len(nums)
        k = 0
    
        for i in range(num):
            if nums[i] != 0:
                nums[k], nums[i] = nums[i], nums[k]
                k += 1
        

File: 283. Move Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {

        // position where next non-zero element should go
        int j = 0;

        // traverse the array
        for (int i = 0; i < nums.size(); i++) {

            // if current element is non-zero
            if (nums[i] != 0) {

                // move non-zero element to position j
                swap(nums[i], nums[j]);

                // next available position
                j++;
            }
        }
    }
};
